CoffeeMachineSimulator/CoffeeMachineSimulator.Services/Services/CoffeeService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using CoffeeMachineSimulator.Services.Interfaces;
+using CoffeeMachineSimulator.Services.Models;
+using FizzWare.NBuilder;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CoffeeMachineSimulator.Services.Services
+{
+    public class CoffeeService : ICoffeeService
+    {
+        private readonly List<CoffeeModel> coffeeModels;
+
+        public CoffeeService()
+        {
+            coffeeModels = Builder<CoffeeModel>.CreateListOfSize(10).Build().ToList();
+        }
+
+        public void AddCoffee(CoffeeModel coffeeToAdd)
+        {
+            if (coffeeToAdd == null) { throw new Exception("You should not add null entries!"); }
+            foreach(CoffeeModel coffee in coffeeModels)
+            { if (coffee.Id == coffeeToAdd.Id) { throw new Exception("Already exists an entry with the same Id"); } }
+            if (!(coffeeToAdd.Id == Guid.Empty || coffeeToAdd.Name == null || coffeeToAdd.Price == 0.0f))
+            {
+                coffeeModels.Add(coffeeToAdd);
+            }
+           
+        }
+
+        public void DeleteCoffee(Guid coffeeId)
+        {
+            if (coffeeId == Guid.Empty) { throw new Exception("The Id given is empty"); }
+            foreach (CoffeeModel coffee in coffeeModels)
```

Comment:
use coffeeModels.First() method to find your entry in the list. There is nothing wrong with using your approach, it just takes more lines of code than it has to https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.first?view=netcore-3.1